test/integration/fuzz/lockup/withdraw.t.sol

```diff
@@ -115,6 +114,89 @@ abstract contract Withdraw_Integration_Fuzz_Test is Integration_Test, Withdraw_I
         assertEq(actualNFTowner, expectedNFTOwner, "NFT owner");
     }
 
+    /// @dev Given enough fuzz runs, all of the following scenarios will be fuzzed:
+    ///
+    /// - Multiple caller addresses.
+    /// - Multiple values for the current time.
+    /// - Multiple values for the withdrawal address.
+    /// - Multiple withdraw amounts.
+    function testFuzz_Withdraw_PublicCaller(
```

Comment:
unfortunately this test doesn't look right, it should be more minimalistic, this much nested code is not a good thing

<details>
<summary>A revised version</summary>

```solidity
    /// @dev Given enough fuzz runs, all of the following scenarios will be fuzzed:
    ///
    /// - Multiple caller addresses.
    function testFuzz_Withdraw_UnknownCaller(address caller)
        external
        whenNotDelegateCalled
        givenNotNull
        whenToNonZeroAddress
        whenWithdrawAmountNotZero
        whenWithdrawAmountNotGreaterThanWithdrawableAmount
    {
        vm.assume(caller != users.sender && caller != users.recipient);

        // Make the fuzzed address the caller in this test.
        changePrank({ msgSender: caller });

        // Simulate the passage of time.
        vm.warp({ timestamp: defaults.WARP_26_PERCENT() });

        // Make the withdrawal.
        lockup.withdraw({ streamId: defaultStreamId, to: users.recipient, amount: defaults.WITHDRAW_AMOUNT() });

        // Assert that the stream's status is still "STREAMING".
        Lockup.Status actualStatus = lockup.statusOf(defaultStreamId);
        Lockup.Status expectedStatus = Lockup.Status.STREAMING;
        assertEq(actualStatus, expectedStatus);

        // Assert that the withdrawn amount has been updated.
        uint128 actualWithdrawnAmount = lockup.getWithdrawnAmount(defaultStreamId);
        uint128 expectedWithdrawnAmount = defaults.WITHDRAW_AMOUNT();
        assertEq(actualWithdrawnAmount, expectedWithdrawnAmount, "withdrawnAmount");
    }
```

</details>

Reply:
> Unfortunately this test doesn't look right,

I agree with you. I found it dirty as well but I failed to figure out a way that could make it look cleaner. 

The revised version only fuzzes `caller`. The expected result is that it must pass in every scenario since `to` is the recipient.

However, my intention was to write a test that could fuzz both `caller` and `to` at the same time and try to find any set of (caller, to) duos that could violate the following law.

- When `caller` is not recipient or approved third party, `to` must be the recipient.
- When `caller` is recipient or approved third party, `to` can be anything.

Shouldn't we fuzz `to` in the revised version as well?

Reply:
> The revised version only fuzzes caller. The expected result is that it must pass in every scenario since to is the recipient.

yes

>However, my intention was to write a test that could fuzz both caller and to at the same time and try to find any set of (caller, to) duos that could violate the following law.
When caller is not recipient or approved third party, to must be the recipient.
When caller is recipient or approved third party, to can be anything.

well, isn't the test name `testFuzz_Withdraw_PublicCaller`? i.e. cases when caller == recipient or approved third party, to must be the recipient are excluded

we may want to modify the last test `testFuzz_Withdraw`

> Shouldn't we fuzz to in the revised version as well?

i don't see the reason why we would want to fuzz the `to` in this test since it needs always to be the recipient to pass the test

Reply:
> cases when caller == recipient or approved third party, to must be the recipient are excluded

Yeah OK, you are right. I agree with you.

We are testing the property that when an unknown caller withdraws to the recipient, recipient receives the stream amount. And in `testFuzz_Withdraw`, We are testing when a recipient withdraws to a random `to` address, `to` receives the stream amount.

> we may want to modify the last test testFuzz_Withdraw

Why?

> i don't see the reason why we would want to fuzz the to address in this test since it needs always to be the recipient to pass the test

I agree we don't need that. We are already covering it in the integration test: `test_RevertWhen_CallerPublic_ToNotRecipient`.